libraries/hotstuff/qc_chain.cpp

```diff
@@ -378,32 +368,30 @@ namespace eosio::hotstuff {
       std::vector<hs_vote_message> msgs;
 
       if (signature_required){
+         //iterate over all my finalizer keys and sign / broadcast for each that is in the schedule
+         std::vector<fc::crypto::blslib::bls_public_key> finalizers = _pacemaker->get_finalizer_keys();
 
-         //iterate over all my finalizers and sign / broadcast for each that is in the schedule
-         std::vector<name> finalizers = _pacemaker->get_finalizers();
-
-         auto mf_itr = _my_producers.begin();
+         auto mfk_itr = _my_finalizer_keys.begin();
 
-         while(mf_itr!=_my_producers.end()){
+         while(mfk_itr!=_my_finalizer_keys.end()){
 
-            auto prod_itr = std::find(finalizers.begin(), finalizers.end(), *mf_itr);
+            auto fin_itr = std::find(finalizers.begin(), finalizers.end(), mfk_itr->first);
 
-            if (prod_itr!=finalizers.end()) {
+            if (fin_itr!=finalizers.end()) {
 
-               hs_vote_message v_msg = sign_proposal(proposal, *prod_itr);
+               hs_vote_message v_msg = sign_proposal(proposal, mfk_itr->second);
 
                fc_tlog(_logger, " === ${id} signed proposal : block_num ${block_num} phase ${phase_counter} : proposal_id ${proposal_id}",
                               ("id", _id)
                               ("block_num", proposal.block_num())
                               ("phase_counter", proposal.phase_counter)
                               ("proposal_id", proposal.proposal_id));
 
-               //send_hs_vote_msg(v_msg);
                msgs.push_back(v_msg);
 
             };
 
-            mf_itr++;
+            mfk_itr++;
```

Comment:
This similarly could be reworked to be simpler now that `_my_finalizer_keys` is a `std::map`.